cyral/resource_cyral_sidecar_listener.go

```diff
@@ -0,0 +1,415 @@
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+
+				},
+				NewResourceData: func() ResourceData { return &SidecarListenerResource{} },
+			}, ReadSidecarListenersConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+				},
+			},
+		),
+		Schema: map[string]*schema.Schema{
+			ListenerIdKey: {
+				Description: "ID of the listener that will be bound to the sidecar.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			SidecarIdKey: {
+				Description: "ID of the sidecar that the listener will be bound to.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			RepoTypesKey: {
+				Description: "List of repository types that the listener supports. Currently limited to one repo type, eg [\"mysql\"]",
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			UnixListenerKey: {
+				Description: "Unix listener settings.",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				// Notice the MaxItems: 1 here. This ensures that the user can only specify one this block.
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						FileKey: {
+							Description: "File in which the sidecar will listen for the given repository.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+					},
+				},
+			},
+			MultiplexedKey: {
+				Description: "Multiplexed listener, defaults to not multiplexing (false). Not supported for all repository types.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			TcpListenerKey: {
+				Description: "tcp listener settings.",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				// Notice the MaxItems: 1 here. This ensures that the user can only specify one this block.
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						HostKey: {
+							Description: "Host in which the sidecar will listen for the given repository. Omit to listen on all interfaces.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						PortKey: {
+							Description: "Port in which the sidecar will listen for the given repository.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+					},
+				},
+			},
+			MysqlSettingsKey: {
+				Description: "MysqlSettings represents the listener settings for a mysql data repository.",
```

Comment:
I think since this is a user-facing description we could list the repo types that each particular setting is supposed to support. For instance, for `MysqlSettingsKey` it also supports `mariadb`, so would be good to list it in the description so that this can be more intuitive to users.

Reply:
fixed